src/store/audioGenerate.ts

```diff
@@ -0,0 +1,169 @@
+import {
+  AudioItem,
+  AudioStoreState,
+  EditorAudioQuery,
+  FetchAudioResult,
+  IEngineConnectorFactoryActionsMapper,
+  SettingStoreState,
+} from "./type";
+import { convertAudioQueryFromEditorToEngine } from "./proxy";
+import { generateUniqueId } from "./utility";
+
+const audioBlobCache: Record<string, Blob> = {};
+
+type Instance = {
+  invoke: IEngineConnectorFactoryActionsMapper;
+};
+
+export async function fetchAudioFromAudioItem(
+  state: AudioStoreState & SettingStoreState,
+  instance: Instance,
+  {
+    audioItem,
+    cacheOnly,
+  }: {
+    audioItem: AudioItem;
+    cacheOnly?: boolean;
+  }
+): Promise<FetchAudioResult> {
```

Comment:
おっしゃるとおり`cacheOnly`は使わない感じになってますね！
`cacheOnly`、消しちゃって良い気がしました！

その場合この関数がfetchする勝手に以外にキャッシュする機構も備えるようになるので、なにかわかるようにしないとですね。
関数名にcacheを加えるか、こんな感じのドキュメントを追加するのはどうでしょう。
```suggestion
/**
 * エンジンで音声を合成する。
 * 音声のキャッシュ機構も備える。
 */
export async function fetchAudioFromAudioItem(
  state: AudioStoreState & SettingStoreState,
  instance: Instance,
  {
    audioItem,
    cacheOnly,
  }: {
    audioItem: AudioItem;
    cacheOnly?: boolean;
  }
): Promise<FetchAudioResult> {
```

Reply:
一案なのですが、`cacheOnly`の隣に`noSaveCache`を追加することを思いつきました。
どのようなオプションが渡せるかという情報が、関数自体がどのような機能を持つかを表すわけです。

それはそうとしてドキュメンテーションはしますね

Reply:
考え方なるほどです！
オプションを用意しておくという考え方と、機能は実際に必要となるまでは追加しないというYAGNI原則の考え方が衝突するかもとちょっと感じました。

Reply:
最終的におっしゃる通りにしました；`cacheOnly`は消去してドキュメンテーションコメントをつけました．